app/packages/looker/src/worker/disk-overlay-decoder.ts

```diff
@@ -74,16 +80,29 @@ export const decodeOverlayOnDisk = async (
     return;
   }
 
+  // if we have an explicit source defined from sample.urls, use that
+  // otherwise, use the path field from the label
+  let source = sources[`${field}.${overlayPathField}`];
+
+  if (typeof overlayCollectionProcessingParams !== "undefined") {
+    // example: for detections, we need to access the source from the parent label
+    // like: if field is "prediction_masks", we're trying to get "predictiion_masks.detections[INDEX].mask"
+    source =
+      sources[
```

Comment:
any concern that the key we're constructing here doesn't exist on `sources`?

Reply:
in which case it'd return undefined. In fact, it won't exist in OSS

Reply:
my eyes told me the `].` deref was outside the key so thought we might be derefing an `undefined`. my mistake 🙏